### ISSUE TITLE
Build a Normal Island fake neighbourhood for development

### ISSUE THREAD

**Issue by erbridge** [geeksforsocialchange/PlaceCal#1478]
## User story

As an owner of PlaceCal I need to be able to roll PlaceCal out anywhere in the world so that I can make everyone's lives better.

As a developer of PlaceCal I need to not need to import the UK's geography and live data to work on the service so that I can work quickly within a smaller context.

## Acceptance criteria

- [ ] Local development happens on Normal Island, a fake geography with a fake geographic hierarchy
- [ ] A set of seed files I can understand by reading then in one sitting has everything I need to be able to have a blank slate geography for Normal Island to work with
- [ ] We seed at least one of every entity type, too
- [ ] We have tests to confirm the service behaves correctly for UK data
- [ ] The service still works with the UK's geography on staging and production
- [ ] There is an illustration of Normal Island we can refer to clearly for development
- [ ] The makefile and readme have been updated to use seed files and not prod db copy

## Follow up work

We then want to seed more interesting partners and all the rest to give us a "lived in" development environment.

## Implementation notes & questions

A side effect of this (and secretly the goal of it) is that we completely decouple the UK's specifics from PlaceCal, so PlaceCal works seamlessly with any geography.

## Implementation plan

(derived from #1947)

- [ ] clear out old seeding code
  - db/seeds and any migration that populates / manipulates data should be blanked
- [ ] generate actual seed data (may happen concurrently with step below)
- [ ] write seeding script
  - [ ] should be easy to use
  - [ ] should reset database completely
- [ ] update documentation

**Comment by kimadactyl** [comment]
Linked to (superceeds?) #1083 and #876

**Comment by r-ferrier** [comment]
Linked to #1748 (completed)

**Comment by ivan-kocienski-gfsc** [comment]
the biggest hurdle for this is that we currently assume the dataset is UK postcodes because we use a UK postcode lookup service to map addresses to neighbourhoods. 

Kim has spoken of Zoning (my word) where we allow for different datasets for different regions). This would allow us to switch postcode resolution service but it would also be a lot of work.

**Comment by kimadactyl** [comment]
yeah agreed - it mgiht be good to open a ticket specificallty for "allow choosing address to neighbourhood resolution mapping service" or something like that?

we are prob also heading towards needing some kind of global config options

**Comment by ivan-kocienski-gfsc** [comment]
Notes and feedback

> Local development happens on Normal Island, a fake geography with a fake geographic hierarchy

This is done but it is super-limiting as it means the local developer cannot create addresses as the address resolution logic needs internet access to work and needs UK postcode data to resolve to. Unless we put in logic to allow for multiple geographies but this is very Dream Yard and not in scope for this milestone.

> A set of seed files I can understand by reading then in one sitting has everything I need to be able to have a blank slate geography for Normal Island to work with

This is done.

> We seed at least one of every entity type, too

For Neighbourhoods, this is done. For other entities like tags, still outstanding.

> We have tests to confirm the service behaves correctly for UK data

This needs to be defined. We have many tests. They do exercize the neighbourhood model.

> The service still works with the UK's geography on staging and production

Okay. This has not changed.

> There is an illustration of Normal Island we can refer to clearly for development

Not applicable if we are not using Normal Island sample data. See first comment above about Local Development.

> The makefile and readme have been updated to use seed files and not prod db copy

The README is complete in terms of provisioning a new PlaceCal instance. The makefile should be looked at to see if it still needs that much logic in it.

The use of Docker, to me, seems like a complication. We are giving people a choice when we should be giving them clear steps and presenting (at the end) alternate options. If people want to use Docker then that should be a whole help-file on its own, imo.

**Comment by katjam** [comment]
Closing as won't fix

**Comment by kimadactyl** [comment]
Not sure why this is closed - yes its complicated but this unblocks not needing all developers to have a copy of prod data. Opening again.
